cvat-ui/src/components/annotation-page/attribute-annotation-workspace/attribute-annotation-sidebar/attribute-editor.tsx

```diff
@@ -256,14 +258,56 @@ interface ListParameters {
     onChange(value: string): void;
 }
 
-function renderList(parameters: ListParameters, keyMap: ShortcutsState['keyMap']): JSX.Element | null {
+function RenderList(parameters: ListParameters): JSX.Element | null {
     const { inputType, values, onChange } = parameters;
-    if (inputType === 'checkbox') {
-        const sortedValues = ['true', 'false'];
-        if (values[0].toLowerCase() !== 'true') {
-            sortedValues.reverse();
+    const { keyMap } = useSelector((state: CombinedState) => state.shortcuts);
+    const sortedValues = ['true', 'false'];
+    const filteredValues = values.filter((value: string): boolean => value !== config.UNDEFINED_ATTRIBUTE_VALUE);
+    const prevValuesRef = useRef<string[] | null>(null);
+
+    useDynamicLabels(componentShortcuts);
```

Comment:
I am not sure I understand what the hook does and why we need it here. Could you please elaborate?

Reply:
This is to reset the shortcuts to their original state once the component unmounts, there is a ref inside because when added to the dependency array it was causing infinite rerenders.

Reply:
I see, let's then rename to `useResetShortcutsOnUnmount`